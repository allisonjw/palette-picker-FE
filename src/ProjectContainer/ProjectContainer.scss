.project-container-main {
    margin: 15px auto;
    column-count: 4;
    column-gap: 1em;
    background-color: pink;
    height: 600px;
}

.section_projectContainer-project {
    // background-color: cadetblue;
    display: flex;
    margin: 20px;
    // width: 300px;
    // text-align: left;
    // border-radius: 5px;
    // border: solid 1px gray;
    // padding: 5px 0px 0px 0px;
    height: 75em;
    // width: 100%;
    flex-wrap: wrap;
}

