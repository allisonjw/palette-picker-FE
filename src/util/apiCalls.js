const baseUrl = 'https://palette-of-colors-picker.herokuapp.com/api/v1';

export const getAllProjects = async () => {
    const response = await fetch(`${baseUrl}/projects`)
    if (!response.ok) {
      throw Error('Unable to get projects. Try again later.')
    }
    const projects = await response.json();
    return projects;
};

export const getProjectById = async (id) => {
    const response = await fetch(`${baseUrl}/projects/${id}`);
    const data = response.json();
    return data;
};

export const addProject = async (project_name) => {
    const options = {
      method: 'POST',
      body: JSON.stringify(project_name),
      headers: {
        'Content-Type': 'application/json'
      }
    };
    const response = await fetch(`${baseUrl}/projects`, options)
    try {
      return response.json()
    } catch {
      throw Error(response.statusText)
    }
};

export const updateProject = async (update, id) => {
    const options = {
      method: 'PATCH',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(update)
    };
    const response = await fetch(`${baseUrl}/projects/${id}`, options);
    try {
      return response.json()
    } catch {
      throw Error(response.statusText)
    }
};

export const deleteProject = async (id) => {
    const options = {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json'
      }
    };
    const response = await fetch(`${baseUrl}/projects/${id}`, options);
    try {
      return response.json()
    } catch {
      throw Error(response.statusText)
    }
};

export const getAllPalettes = async () => {
    const response = await fetch(`${baseUrl}/palettes`)
    if (!response.ok) {
      throw Error('Unable to get palettes. Try again later.')
    }
    const palettes = await response.json();
    return palettes;
};

export const getPaletteById = async (id) => {
    const response = await fetch(`${baseUrl}/palettes/${id}`);
    const data = response.json();
    return data;
};

export const addPalette = async (palette) => {
    const options = {
      method: 'POST',
      body: JSON.stringify(palette),
      headers: {
        'Content-Type': 'application/json'
      }
    };
    const response = await fetch(`${baseUrl}/palettes`, options)
    try {
      return response.json()
    } catch {
      throw Error(response.statusText)
    }
};

export const updatePalette = async (update, id) => {
    const options = {
      method: 'PATCH',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(update)
    };
    const response = await fetch(`${baseUrl}/palettes/${id}`, options);
    try {
      return response.json()
    } catch {
      throw Error(response.statusText)
    }
};

export const deletePalette = async (id) => {
    const options = {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json'
      }
    };
    const response = await fetch(`${baseUrl}/palettes/${id}`, options);
    try {
      return response.json()
    } catch {
      throw Error(response.statusText)
    }
};
  
  