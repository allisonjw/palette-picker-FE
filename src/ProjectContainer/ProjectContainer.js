import Project from '../Project/Project';
import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { setProjects, deleteProject } from '../actions';
import { getAllProjects } from '../util/apiCalls';
import './ProjectContainer.scss';

export const ProjectContainer = () => {
    const dispatch = useDispatch();
    const [project, setAllProjects] = useState([]);
    const projects = useSelector(state => state.projects)

        const fetchProjects = async () => {
          try {
            const response = await getAllProjects();
            setAllProjects(response.projects);
            dispatch(setProjects(response));
          } catch(error) {
              console.log(error);
          }
        };
      useEffect(() => {
        fetchProjects();
      }, []);

      return(
          <section className="section_projectContainer">
              <Project projects={projects}/>
          </section>
      )
    
    
}

export default ProjectContainer;
