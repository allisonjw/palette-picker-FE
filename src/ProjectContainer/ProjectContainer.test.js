import React from 'react';
import { render } from '@testing-library/react';
import ProjectContainer from './ProjectContainer';
import { shallow } from 'enzyme';
import { getAllProjects, getAllPalettes } from '../util/apiCalls';

jest.mock('../util/apiCalls');

jest.mock("react-redux", () => ({
  useDispatch: () => [{}],
  useSelector: () => ({ projects : [{}, {}] })
}));

describe('ProjectContainer', () => {
  let wrapper;
  const mockProjects =[{
        "id": 9,
        "project_name": "Jessie's House",
        "user_id": 5
    },
    {
        "id": 10,
        "project_name": "Dream Bath",
        "user_id": 5
    },
    {
        "id": 11,
        "project_name": "Allison's House",
        "user_id": 6    
  }];

  let mockPalettes = [{
        "id": 9,
        "palette_name": "Neutrals",
        "project_id": 10,
        "color_1": "#D6B28D",
        "color_2": "#C1A281",
        "color_3": "#E5CCA9",
        "color_4": "#BFA98B",
        "color_5": "#A88F77",
    },
    {
        "id": 10,
        "palette_name": "Sailing Away",
        "project_id": 9,
        "color_1": "#17E2B4",
        "color_2": "#143472",
        "color_3": "#060E1E",
        "color_4": "#A3AFC6",
        "color_5": "#E52522",
    },
    {
        "id": 11,
        "palette_name": "Modern",
        "project_id": 11,
        "color_1": "#AE2D49",
        "color_2": "#21354A",
        "color_3": "#4EB47A",
        "color_4": "#C9CFC8",
        "color_5": "#F6C876",
    },
    {
        "id": 12,
        "palette_name": "Mountain Dreams",
        "project_id": 12,
        "color_1": "#C6CDB7",
        "color_2": "#ECB78A",
        "color_3": "#5A7673",
        "color_4": "#BBB493",
        "color_5": "#96B2B4",
    },
    {
        "id": 24,
        "palette_name": "Murphy's",
        "project_id": null,
        "color_1": "#4afb62",
        "color_2": "#431960",
        "color_3": "#901066",
        "color_4": "#9ab3b9",
        "color_5": "#9cd06e",
    }];

  beforeEach(() => {
    wrapper = shallow(<ProjectContainer projects={mockProjects} palettes={mockPalettes}/>)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});