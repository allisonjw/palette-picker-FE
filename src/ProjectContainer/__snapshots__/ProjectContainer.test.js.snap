// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectContainer should match the snapshot 1`] = `
<div
  className="project-container-main"
>
  <section
    className="section_projectContainer-project"
  >
    <Project
      projects={
        Object {
          "projects": Array [
            Object {},
            Object {},
          ],
        }
      }
    />
  </section>
</div>
`;
